src/modules/linux-micro/automount/automount.c

```diff
@@ -0,0 +1,347 @@
+        return false;
+
+    if ((sb->s_feature_ro_compat & EXT3_FEATURE_RO_COMPAT_UNSUPPORTED) ||
+        (sb->s_feature_incompat & EXT3_FEATURE_INCOMPAT_UNSUPPORTED))
+        return false;
+
+    return true;
+}
+
+static bool
+ext4_probe_cb(int fd)
+{
+    char buff[512];
+    struct ext_super_block *sb;
+
+    if (!ext_read_superblock(fd, buff))
+        return false;
+
+    sb = (struct ext_super_block *)buff;
+    if (sb->s_feature_incompat & EXT3_FEATURE_INCOMPAT_JOURNAL_DEV)
+        return false;
+
+    return true;
+}
+
+static const struct filesystem table[] = {
+    { .id = "vfat", .magics = vfat_magic },
+    { .id = "hfsplus", .magics = hfsplus_magic },
+    { .id = "hfs", .magics = hfs_magic },
+    { .id = "ext2", .magics = ext_magic, .fs_cb = ext2_probe_cb },
+    { .id = "ext3", .magics = ext_magic, .fs_cb = ext3_probe_cb },
+    { .id = "ext4", .magics = ext_magic, .fs_cb = ext4_probe_cb },
+};
+
+static bool
+automount_test_magic(const struct filesystem *fs, const struct filesystem_magic *magic, char *dev)
+{
+    int fd, err;
+    char buff[magic->length];
+    bool res;
+
+    fd = open(dev, O_RDONLY | O_CLOEXEC);
+    if (fd < 0) {
+        SOL_ERR("Could not open dev: %s", dev);
+        return false;
+    }
+
+    err = lseek(fd, magic->offset, SEEK_SET);
+    if (err < 0) {
+        SOL_ERR("Coud not lseek dev: %s", dev);
+        goto err;
+    }
+
+    err = read(fd, buff, magic->length);
+    if (err != magic->length) {
+        SOL_ERR("Could not read dev: %s", dev);
+        goto err;
+    }
+
+    res = streq(buff, magic->magic);
+    if (res && fs->fs_cb) {
+        res = fs->fs_cb(fd);
+    }
+
+    close(fd);
+    return res;
+
+err:
+    close(fd);
+    return false;
+}
+
+static const char *
+automount_get_fstype(char *dev)
+{
+    unsigned int i;
+
+    for (i = 0; i < ARRAY_SIZE(table); i++) {
+        const struct filesystem_magic *magic;
+
+        for (magic = table[i].magics; magic->magic != NULL; magic++) {
+            if (automount_test_magic(&table[i], magic, dev))
+                return table[i].id;
+        }
+    }
+
+    return NULL;
+}
+
+static void
+mount_async_cb(const char *dev, const char *mpoint, void *data, uint64_t pid, int status)
+{
+    if (status == 0)
+        SOL_INF("Successfully auto-mounted %s to %s", dev, mpoint);
+    else
+        SOL_ERR("Could not auto-mount %s to %s", dev, mpoint);
+}
+
+static void
+uevent_cb(void *data, struct sol_uevent *uevent)
+{
+    char dev[PATH_MAX], mpoint[PATH_MAX];
+    const char *fstype;
+    int err;
+
+    if (!sol_str_slice_str_eq(uevent->devtype, "partition")) {
+        SOL_DBG("Non partition devtype (%s), skipping", uevent->devtype.data);
+        return;
+    }
+
+    err = snprintf(dev, sizeof(dev), "/dev/%s", uevent->devname.data);
+    if (err < 0 || err >= (int)sizeof(dev)) {
+        SOL_ERR("Could not format nodedev path");
+        return;
+    }
+
+    err = snprintf(mpoint, sizeof(mpoint), "/mnt/%s", uevent->devname.data);
```

Comment:
Do we want to mount everything under `/mnt`, or `/media` or other mount points are also valid?